Realm/ObjectStore/handover.cpp

```diff
@@ -0,0 +1,190 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// This is lets our internals access AnyHandover's private constructors.
+#define private public
+#include "handover.hpp"
+#undef private
+
+using namespace realm;
+
+#pragma mark AnyThreadConfined
```

Comment:
No `#pragma mark` in object store files (it's a non-standard extension and the OS code isn't clang-only like the obj-c code is).